### PR TITLE
HDDS-4223. Optimize the construction method of OzoneVolume and OzoneBucket.

### PR DESCRIPTION
**Pull request by whbing**
## What changes were proposed in this pull request?

Optimize the construction method of OzoneVolume and OzoneBucket.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4223

### CODE REVIEW

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneBucket.java [hunk 1]
```diff
@@ -1034,6 +889,135 @@ public boolean setOwner(String userName) throws IOException {
...
+  /**
+   * Inner builder for OzoneBucket.
+   */
+  public static class Builder {
+    private Map<String, String> metadata;
+    private ConfigurationSource conf;
+    private ClientProtocol proxy;
+    private String volumeName;
+    private String name;
+    private DefaultReplicationConfig defaultReplicationConfig;
```
Comment (whbing) [inline hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneBucket.java:907]: Note: The `DefaultReplicationConfig` type from the original constructor is retained. `ReplicationConfig` is also available.

**Comment by kerneltime** [comment]
@sumitagrawl can you please take a look as well?

**Review by sumitagrawl — APPROVED** [review]
LGTM +1

**Review by captainzmc — APPROVED** [review]
+1 The change looks good.  Thanks @whbing for the patch.

**Pull request merged**